provider/core/src/constructors.rs

```diff
@@ -99,3 +111,271 @@
 //! [`StaticDataProvider`]: ../../icu_provider_blob/struct.StaticDataProvider.html
 //! [`FsDataProvider`]: ../../icu_provider_blob/struct.FsDataProvider.html
 //! [`ICU4XDataProvider`]: ../../icu_capi/provider/ffi/struct.ICU4XDataProvider.html
+
+#[doc(hidden)]
+#[macro_export]
+macro_rules! gen_any_buffer_unstable_docs {
```

Comment:
Nit: I do like moving these macros into this file, but I would prefer if you would do that in a standalone PR so that I can review the diffs better.

Reply:
This is not a move, I'm adding a different version of the macro so we can migrate crates one by one.